raiden-ts/src/services/epics.ts

```diff
@@ -320,43 +323,50 @@ export function pfsFeeUpdateEpic(
 ): Observable<messageGlobalSend.request> {
   return state$.pipe(
     groupChannel$,
-    // get only first state per channel
-    mergeMap((grouped$) => grouped$.pipe(first())),
-    withLatestFrom(config$),
-    // ignore actions while/if mediating not enabled
-    filter(
-      ([channel, { pfsRoom, caps }]) =>
-        channel.state === ChannelState.open && !!pfsRoom && !!getCap(caps, Capabilities.MEDIATE),
-    ),
-    mergeMap(([channel, { pfsRoom }]) => {
-      const message: PFSFeeUpdate = {
-        type: MessageType.PFS_FEE_UPDATE,
-        canonical_identifier: {
-          chain_identifier: BigNumber.from(network.chainId) as UInt<32>,
-          token_network_address: channel.tokenNetwork,
-          channel_identifier: BigNumber.from(channel.id) as UInt<32>,
-        },
-        updating_participant: address,
-        timestamp: makeTimestamp(),
-        fee_schedule: {
-          cap_fees: true,
-          imbalance_penalty: null,
-          proportional: Zero as Int<32>,
-          flat: Zero as Int<32>,
-        },
-      };
-      const msgId = makeMessageId().toString();
+    mergeMap((grouped$) =>
+      combineLatest([grouped$, config$]).pipe(
+        filter(([, { pfsRoom, caps }]) => !!pfsRoom && !!getCap(caps, Capabilities.MEDIATE)),
+        map(([channel, config]) => {
+          const feeSchedule = getFeeScheduleForChannel(channel, config);
+          // using channel feeSchedule above, build a PFSFeeUpdate's schedule payload
+          const schedule: PFSFeeUpdate['fee_schedule'] = {
+            cap_fees: true,
+            flat: feeSchedule.flat,
```

Comment:
As everywhere before: do we need to be so specific here about our models and the configuration parameter names? It is an absolute nightmare to extend and maintain this feature.

Reply:
I don't like your usage of `nightmare` all over the place. This is a HARDCODED feature, and in no place was it specified that it had a different requirement. Extensibility was achieved by having self-contained, pure isolated functions when possible, but you still need to write what you want it to do. If you want us to introduce a plugin system, we should have calls with management do decide on that, but I didn't expect nor want to do this on this feature/PR.